ss_admin_console/main/default/classes/SSACFilter.cls

```diff
@@ -1,162 +1,144 @@
+            for (SObjectField nonKeyField: nonKeyFields) {
+                    if (record.get(nonKeyField) == existentMap.get(concatenatedString).get(nonKeyField)) {
+                        matchedNonKeyFields += 1;
+                    } else {
+                        results.add(new SSACResult()
+                            .setStatus(SSACConstants.CHANGED_STATUS)
+                            .setRecordKey(concatenatedString)
+                            .setRecordId(existentMap.get(concatenatedString).Id)
+                            .setSObjectApiName(String.valueOf(sObjectApiName)));
+                        break;
+                    }
+                }
+                if (matchedNonKeyFields == nonKeyFields.size()) {
+                    results.add(new SSACResult()
+                    .setStatus(SSACConstants.DUPLICATED_STATUS)
+                    .setRecordKey(concatenatedString)
+                    .setSObjectApiName(String.valueOf(sObjectApiName)));
+                }
+            } else {
+                results.add(new SSACResult()
+                .setStatus(SSACConstants.OK_STATUS)
+                .setRecordKey(concatenatedString)
+                .setSObjectApiName(String.valueOf(sObjectApiName)));
+            }
+        }
+        return results;
+    }
+
+    @TestVisible
+    private Map<String, SObject> generateExistentMap(){
+        Map<String, SObject> existentMap = new Map<String, SObject>();
+        for (SObject existSObj: existentSObjects) {
+            String concatenatedString = '';
+            for (SObjectField keyField: keyFields) {
+                if (String.valueof(keyField).contains('.')) {
+                    List<String> splittedField = String.valueof(keyField).split('\\.');
+                    String nextField = splittedField[1];
+                    concatenatedString += String.valueOf(existSObj.getSObject(splittedField[0]).get(nextField)).toLowerCase() + '_';	
+                } else {
+                    concatenatedString += String.valueOf(existSObj.get(String.valueof(keyField))).toLowerCase() + '_';	
+                }
+            }
+            concatenatedString = concatenatedString.removeEnd(concatenatedString.right(1));
+            existentMap.put(concatenatedString, existSObj);
+        }
+        return existentMap;
+    }
 
-	@TestVisible
-	private List<SObject> getExistentSObjectsv2(){
+    // existent sobject extraction
+    @TestVisible
+    private List<SObject> getExistentSObjects() {
         String query = this.buildQuery();
-		Map<String, SObject> result = new Map<String, SObject>();
-        List<SObject> existentRecords = Database.query(query);
-        return existentRecords;
+        return Database.query(query);
     }
 
-	@TestVisible
-    private Map<String, Set<Object>> getConstantFields(){
+    // formatting map with all fields and its values
+    @TestVisible
+    private Map<String, Set<Object>> getConstantFields() {
```

Comment:
Does the method return fields or values? Why constants, I'd rename it to getKeyFieldValues